addons/web/static/src/core/assets.js

```diff
@@ -108,23 +119,58 @@ export const loadBundleTemplates = memoize(async function loadBundleTemplates(na
     return processTemplates(templates);
 });
 
+
+/**
+ * Loads the content definition of a bundle
+ * @param {string} name the name of the bundle as declared in the manifest.
+ * @returns {Promise<BundleInfo[]>} A promise of the content definition of the bundle
+ * @private
+ */
+const _loadBundle = memoize(async function (bundleName) {
+    const bundleURL = new URL(`/web/bundle/${bundleName}`, window.location.origin);
+    const request = await browser.fetch(bundleURL.href, {
+        method: "get",
+        header: {
+            "Content-Type": "application/json"
+        },
+        param: JSON.stringify({}),
+    })
+    return await request.json();
+});
+
 const bundlesCache = {};
+
+
 /**
  * Loads a bundle.
  *
  * @param {string} name the name of the bundle to load
  * @param {Bundle} options parts of the bundle to load (see Bundle typedef)
  * @returns {Promise<LoadedBundle>}
  */
-async function loadBundle(name, options) {
+export async function loadBundle(name, options) {
     const { templates } = options;
     if (!bundlesCache[name]) {
         bundlesCache[name] = { name };
     }
     if (templates && !bundlesCache[name].templates) {
         bundlesCache[name].templates = loadBundleTemplates(name);
     }
-    // TODO: if ("js/css") {...} to support lazy loading js/css from bundles
+
+    if (options.css && !bundlesCache[name].css || options.js && !bundlesCache[name].js) {
+        // we only load the bundle info (that is a RPC) if JS or CSS is requested on the bundle,
+        // and if it has not yet been requested.
+        /** @type BundleInfo[] */
```

Comment:
Any idea why we need this type declaration? The `_loadBundle` function seems correctly typed so the type should be inferred from the call.

Reply:
pycharms prefers it